### DISCUSSION TITLE
Why does the __getitem__ function applies transform when the passed index is only int

### DISCUSSION THREAD

**Discussion by yasinzaii** [Project-MONAI/MONAI#7739] [Q&A]
The __getitem__ function should be returning the desired data samples. Which if does if the desired index passed in slice and sequence. But if the desired index is integer it apply transform on top of it. Is there any special reason for it ?

**Comment by KumoLiu** [comment]
Hi @yasinzaii , thanks for your interest here.
The `index` can also be slice or sequence which help you get a subset of the data.
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset.py#L100

Thanks.

**Comment by yasinzaii** [comment]
I know, but why apply transform when a single integer idex is pass as argument. Whereas on the other hand just a subset is passed without a tansform. What is the reason of that

**Reply by KumoLiu** [reply]
Hi @yasinzaii, it only returns a subset when you pass a slice. Therefore, you can still iterate through the subset to obtain the transformed data. And in MONAI, we usually assume that the input is a numpy or tensor for the transforms.
